File: frontend-office/mobile-flutter/lib/models/sensor_reading.dart
class SensorReading {
  final int id;
  final String deviceId;
  final double voltage;
  final double? temperature;
  final double? humidity;
  final bool relayOn;
  final DateTime recordedAt;

  SensorReading({
    required this.id,
    required this.deviceId,
    required this.voltage,
    this.temperature,
    this.humidity,
    required this.relayOn,
    required this.recordedAt,
  });

  factory SensorReading.fromJson(Map<String, dynamic> json) {
    return SensorReading(
      id: json['id'] as int,
      deviceId: json['device_id'] as String,
      voltage: (json['voltage'] as num).toDouble(),
      temperature: json['temperature'] != null
          ? (json['temperature'] as num).toDouble()
          : null,
      humidity: json['humidity'] != null
          ? (json['humidity'] as num).toDouble()
          : null,
      relayOn: json['relay_on'] as bool? ?? false,
      recordedAt: DateTime.parse(json['recorded_at'] as String),
    );
  }
}

File: backend-admin-spring/src/main/java/com/smartwind/admin/model/SensorReading.java
package com.smartwind.admin.model;

import jakarta.persistence.*;
import java.time.OffsetDateTime;

@Entity
@Table(name = "sensor_readings")
public class SensorReading {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "device_id", nullable = false, length = 64)
    private String deviceId;

    @Column(nullable = false)
    private Double voltage;

    private Double temperature;

    private Double humidity;

    @Column(name = "relay_on")
    private Boolean relayOn;

    @Column(name = "recorded_at", nullable = false)
    private OffsetDateTime recordedAt;

    public Long getId() { return id; }
    public String getDeviceId() { return deviceId; }
    public Double getVoltage() { return voltage; }
    public Double getTemperature() { return temperature; }
    public Double getHumidity() { return humidity; }
    public Boolean getRelayOn() { return relayOn; }
    public OffsetDateTime getRecordedAt() { return recordedAt; }
}
